src/types/UiaTextRangeBase.cpp

```diff
@@ -1450,6 +1504,178 @@ std::pair<Endpoint, Endpoint> UiaTextRangeBase::_moveByCharacterBackward(gsl::no
+            auto target = _endpointToCoord(pData, point);
+            buffer.GetSize().IncrementInBounds(target);
+
+            target = (moveState.Increment == MovementIncrement::Forward) ?
+                         buffer.GetWordEnd(target, wordDelimiters, true) :
+                         buffer.GetWordStart(target, wordDelimiters, true);
+
+            currentColumn = target.X;
+        }
+        *pAmountMoved += static_cast<int>(moveState.Increment);
+
+        FAIL_FAST_IF(!(currentColumn >= _getFirstColumnIndex()));
+        FAIL_FAST_IF(!(currentColumn <= _getLastColumnIndex(pData)));
+        FAIL_FAST_IF(!(currentScreenInfoRow >= _getFirstScreenInfoRowIndex()));
+        FAIL_FAST_IF(!(currentScreenInfoRow <= _getLastScreenInfoRowIndex(pData)));
+    }
+
+    Endpoint end = _screenInfoRowToEndpoint(pData, currentScreenInfoRow) + currentColumn;
+
+    auto target = _endpointToCoord(pData, end);
+    target = buffer.GetWordStart(target, wordDelimiters, true);
+    Endpoint start = _coordToEndpoint(pData, target);
+
+    return std::make_pair<Endpoint, Endpoint>(std::move(start), std::move(end));
+}
+
+std::pair<Endpoint, Endpoint> UiaTextRangeBase::_moveByWordBackward(gsl::not_null<IUiaData*> pData,
+                                                                    const int moveCount,
+                                                                    const MoveState moveState,
+                                                                    const std::wstring& wordDelimiters,
+                                                                    _Out_ gsl::not_null<int*> const pAmountMoved)
+{
+    // STRATEGY:
+    //    - move the "start" Endpoint to the proper place (if need to move multiple times, do it here)
+    //    - find the "end" Endpoint based on that
+
+    *pAmountMoved = 0;
+    const int count = moveCount;
+    ScreenInfoRow currentScreenInfoRow = moveState.StartScreenInfoRow;
+    Column currentColumn = moveState.StartColumn;
+
+    auto& buffer = pData->GetTextBuffer();
+    for (int i = 0; i < abs(count); ++i)
+    {
+        // check if we're at the edge of the screen info buffer
+        if (currentScreenInfoRow == moveState.LimitingRow &&
+            currentColumn == moveState.LastColumnInRow)
+        {
+            break;
+        }
+        else if (currentColumn == moveState.LastColumnInRow)
+        {
+            // we're at the edge of a row and need to go to the
+            // previous one. move to the cell with the last non-whitespace charactor
+
+            currentScreenInfoRow += static_cast<int>(moveState.Increment);
+
+            // get the right-most char for the previous row
+            const ROW& row = buffer.GetRowByOffset(currentScreenInfoRow);
+            const size_t right = row.GetCharRow().MeasureRight();
+            currentColumn = gsl::narrow<Column>((right == 0) ? 0 : right - 1);

```

Comment:
We tend to try to write things like `(right == 0) ? 0 : right - 1` instead as `std::max(0, right - 1)`.

Reply:
Dumb problem:
- `right-1` is a `size_t` (so, `unsigned long long`)
- 0 can't be `static_cast` into a `size_t`
Can you think of a workaround?

Reply:
@miniksa I'm not sure that is a valid transformation, given the unsignedness of `right`. The original check is `right==0`, and unsigned overflow _IS DEFINED_, so `right-1` would be `ULONGLONG_MAX` if it was 0... `max` would really mess us up.

Reply:
OK.